### DISCUSSION TITLE
ipfkey

### DISCUSSION THREAD

**Discussion by maguag** [mtex-toolbox/mtex#1701] [Ask Anything] [answered]
Dear Developers：
    I am trying to change the orientation of ipfkey, it seems that default ipfkey based on （001）（-111）and （011）, thus i want to know how to change it to （001） （111）and（101）?

![e8faf3800dfb7e3fc42b7d53e15653a](https://user-images.githubusercontent.com/48279761/209915653-137f97de-21d0-4766-aa41-dbd91adcd1d8.png)
.

```matlab
cs = crystalSymmetry('m-3m');
ipfKey = ipfColorKey(cs);
plot(ipfKey);
```

**Comment by kilir** [comment, marked as answer]
Hi,
please have a quick search here in the forum, if I remember correctly, several people already had the desire to change the labeling as well.
Cheers,
Rüdiger

**Comment by maguag** [comment]
Thanks, I have found the disscussion and successfuly changed the lables to （001） （111）and（101) by modify the codes as follow：
```matlab
cs = crystalSymmetry('m-3m');

ipfkey= ipfColorKey(cs);
h= Miller({0 0 1},{1 0 1},{1 1 1},cs);
f= sphericalRegion.byVertices(h);

plot(ipfkey,f);
```
![untitled](https://user-images.githubusercontent.com/48279761/210293281-f86d430c-6223-4410-90d8-23aa5e888f40.jpg)

**Reply by ddyyasfh** [reply]
Hello, may I ask how the polar chart drawn above ?

**Reply by maguag** [reply]
Hi, run the follow code,  and you will get the figure above.
```matlab
mtexdata copper
ebsd('copper').CS = ebsd('c').CS.Laue;
figure
ipfKey = ipfHSVKey(ebsd('copper'));
plot(ipfKey,'complete','upper');
h = Miller({1,0,0},{1,1,1},{1,0,1},{0,1,1},ebsd('copper').CS);
annotate(h.symmetrise,'labeled','backgroundColor','w')
```

**Reply by ddyyasfh** [reply]
thank you very much
